vmdb/app/models/ems_microsoft/powershell.rb

```diff
@@ -34,11 +34,14 @@ def powershell_xml_to_hash(xml)
         MiqPowerShell::Convert.new(xml).to_h
       end
 
-      def log_dos_error_results(results)
+      def log_dos_error_results(error)
```

Comment:
With this change, I think that line 15 of this file (in `run_powershell_script`) needs to change, too.  Otherwise, it looks like it will log an error on any result.

In fact, now that `ScvmmErrorHandling` exists, it might make sense to just move `log_dos_error_results` and `extract_errors_from_results` to `ScvmmErrorHandling`.

Then, the call could just be:

```ruby
results = connection.run_cmd(command)
ScvmmErrorHandling.raise_error_condition(results)
```

And, then `ScvmmErrorHandling.raise_error_condition` could call `extract_errors_from_results`, `log_dos_error_results`, and raise the appropriate error, if necessary.